agent/recorder-agent.c

```diff
@@ -249,6 +274,218 @@ gboolean
+    if (arg_edge_id && g_strcmp0 (arg_edge_id, "")
+        && g_strcmp0 (arg_edge_id, edge_id))
+      continue;
+    if (!g_file_test (recording_edge_dir, G_FILE_TEST_IS_DIR))
+      continue;
+    if (subdir) {
+      g_dir_close (subdir);
+      subdir = NULL;
+    }
+    if (error) {
+      g_error_free (error);
+      error = NULL;
+    }
+    subdir = g_dir_open (recording_edge_dir, 0, &error);
+    if (error)
+      continue;
+    while ((filename = g_dir_read_name (subdir))) {
+      g_clear_pointer (&record_id, g_free);
+      g_clear_pointer (&file_id, g_free);
+      check_filename (filename, arg_from, arg_to, &record_id, &file_id);
+      if (!record_id || !file_id)
+        continue;
+
+      if (arg_record_id && g_strcmp0 (arg_record_id, "")
+          && g_strcmp0 (arg_record_id, record_id))
+        continue;
+      records_found = TRUE;
+
+      g_clear_pointer (&filename_full, g_free);
+      filename_full = g_build_filename (recording_edge_dir, filename, NULL);
+      g_stat (filename_full, &st);
+
+      if (!arg_edge_id)
+        g_variant_builder_add (builder, "(sxxx)", g_strdup (record_id), 0, 0,
+            st.st_size);
+      else if (!arg_record_id)
+        g_variant_builder_add (builder, "(ssxxx)", g_strdup (record_id),
+            g_strdup (file_id), 0, 0, st.st_size);
+    }
+  }
+
+  if (dir)
+    g_dir_close (dir);
+  if (subdir)
+    g_dir_close (subdir);
+  g_clear_pointer (&record_id, g_free);
+  g_clear_pointer (&file_id, g_free);
+  g_clear_pointer (&recording_edge_dir, g_free);
+  g_clear_pointer (&filename_full, g_free);
+
+  if (error)
+    g_error_free (error);
+
+  return edge_id;
+}
+
+gboolean
+    hwangsae_recorder_agent_recorder_interface_handle_lookup_by_record
+    (Hwangsae1DBusRecorderInterface * object,
+    GDBusMethodInvocation * invocation, gchar * arg_record_id, gint64 arg_from,
+    gint64 arg_to, gpointer user_data) {
+  HwangsaeRecorderAgent *self = (HwangsaeRecorderAgent *) user_data;
+  g_autofree gchar *cmd = NULL;
+  g_autofree gchar *response = NULL;
+  g_autofree gchar *recording_dir;
+  g_autofree gchar *edge_id = NULL;
+  GVariantBuilder *builder;
+  GVariant *records;
+
+  g_debug
+      ("hwangsae_recorder_agent_recorder_interface_handle_lookup_by_record");
+
+  recording_dir = g_settings_get_string (self->settings, "recording-dir");
+
+  builder = g_variant_builder_new (G_VARIANT_TYPE ("a(sxxx)"));
+
+  edge_id = get_records (recording_dir, NULL, arg_record_id, arg_from, arg_to,
+      builder);
+
+  records = g_variant_new ("a(sxxx)", builder);
+
+  g_variant_builder_unref (builder);
+
+  hwangsae1_dbus_recorder_interface_complete_lookup_by_record (object,
+      invocation, edge_id, records);
+
+  return TRUE;
+}
+
+gboolean
+    hwangsae_recorder_agent_recorder_interface_handle_lookup_by_edge
+    (Hwangsae1DBusRecorderInterface * object,
+    GDBusMethodInvocation * invocation, gchar * arg_edge_id, gint64 arg_from,
+    gint64 arg_to, gpointer user_data) {
```

Comment:
Yeah that commit hook checker sometimes forces you into hideous formatting like this. I think no one will complain if you override it in such cases. `git commit --no-verify` will skip the hook and let you use any format your like.

```
gboolean
hwangsae_recorder_agent_recorder_interface_handle_lookup_by_edge (
    Hwangsae1DBusRecorderInterface * object,
    GDBusMethodInvocation * invocation, gchar * arg_edge_id, gint64 arg_from,
    gint64 arg_to, gpointer user_data)
{
```
I think the above is much better (i.e. not weird). Also you can make it static.